### PR TITLE
Add get_transcript tool to read creator conversation history

### PR DESCRIPTION
**Pull request by gtanczyk**
## Summary

Adds a new `get_transcript` MCP tool that allows agents to read the creator conversation history across the current build round and up to five earlier sibling rounds. This fixes a critical issue where terse follow-up messages (e.g., "build my game plz") were treated as the complete request, causing agents to miss important context from earlier messages in the conversation.

## Key Changes

- **New `build-transcript.ts` module**: Implements `loadBuildTranscript()` to serve creator conversation in windowed pages rather than all at once. Features:
  - Returns the tail (most recent window) by default with `DEFAULT_TRANSCRIPT_WINDOW_ENTRIES = 20`
  - Supports pagination via `cursor` and `limit` parameters
  - Enforces per-window byte ceiling (`MAX_TRANSCRIPT_WINDOW_BYTES = 20 KB`) by shrinking window size, never dropping entries out of order
  - Strips playtest instrumentation context and hides pre-cutover presence artifacts
  - Merges creator messages and build events across sibling rounds, oldest-first

- **Agent channel integration**: Added `GET /api/agent/build/transcript` endpoint that returns transcript entries with `hasMore` and `nextCursor` for pagination

- **MCP server integration**: Registered `get_transcript` as a visible MCP tool with proper documentation in the workflow

- **Session nudging**: Added `transcript_unread` warning that appears on rounds > 1 (revision/resumed rounds) until `get_transcript` is called, with a reminder limit to avoid noise

- **Build prompt updates**: Removed inline history block and last-message inlining from revision round prompts. Now points agents to `read_inbox` and `get_transcript` instead, with guidance that the latest message is the tail of a conversation, not the whole of it

- **Moved shared utilities**: Extracted `stripPlaytestContext()` and `PLAYTEST_CONTEXT_HEADER` from `submissions.ts` to `build-transcript.ts` for reuse

## Notable Implementation Details

- Windowing approach prevents token ceiling issues that plagued the previous whole-conversation approach (similar to `get_kit_api` refusing at 100 KB)
- Cursor values are clamped rather than erroring on malformed/stale values, degrading gracefully to sensible defaults
- Playtest context blocks are stripped from creator-facing surfaces but preserved in agent inbox messages where they provide needed instrumentation data
- Presence event leftovers from before the cutover are hidden from transcripts to avoid noise
- Comprehensive test coverage for pagination, byte limits, multi-round merging, and edge cases

https://claude.ai/code/session_015gb6VTBYsrhH5C17S9PStb

### CODE REVIEW

**Comment by gtanczyk** [comment]
@codex

File: apps/api/src/build-prompt.ts [hunk 1]
```diff
@@ -138,17 +119,45 @@ function channelDelivery(brief: BuildBrief, creating: boolean): string[] {
...
 }
 
-// The creator's own words, fenced, last — plus anything a caller appended.
+// The creator's words, last. A fresh round's spec is inlined; a revision round is
+// pointed at the channel instead. Inlining the last relayed message looked complete and
+// wasn't: when a round dies before its agent reads the real request, the next round
+// holds only the creator's follow-up nudge, and "build my game plz" fenced as *the*
+// request reads as the whole ask while the full spec sits unread in the brief.
 function finish(lines: string[], brief: BuildBrief): string {
+  if (brief.feedback || !brief.spec.trim()) {
```
[the remark this answers is not in the document]
Comment (gtanczyk) [reply apps/api/src/build-prompt.ts:141]: Confirmed and fixed in 1dc2568. `resumeBuild` now takes `feedbackQueueFailed`, set by the feedback route when `appendCreatorMessage` throws but the request still falls through to dispatch (the in-flight-round path already fails closed on this; this was the other one). `buildPrompt` inlines the feedback text directly in that one case instead of pointing at `read_inbox`/`get_transcript`, which would have nothing to serve. Covered by a new test in `submissions.test.ts` that fails the queue write and asserts the dispatched brief still carries the text.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/api/src/mcp-session-nudges.ts [hunk 2]
```diff
@@ -184,6 +204,16 @@ export function createMcpNudgeTracker(
     state.seedStatus = status;
   }
 
+  function noteRoundGeneration(jobId: number, round: number, nowMs: number): void {
+    const state = ensure(jobId, nowMs);
```
[the remark this answers is not in the document]
Comment (gtanczyk) [reply apps/api/src/mcp-session-nudges.ts:203]: Confirmed and fixed in 1dc2568. Replaced `noteRoundGeneration`/`round` with `noteDispatchAttempt`/`dispatchAttempt` (backed by `dispatch.refs.length`, which also fixed a separate gap: `round` alone missed an undelivered retry, since `ensureRoundGeneration` doesn't bump it). `noteDispatchAttempt` now compares against the last-seen value and resets `transcriptFetched`/`transcriptRemindersSent` whenever it changes, exactly as suggested. New tests in `mcp-session-nudges.test.ts` cover both the re-arm-on-change and no-re-arm-on-same-value cases.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/api/src/build-transcript.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    : [];
+  const rounds = [
+    { record, round: 'current' as const },
+    ...siblings.map((sibling) => ({ record: sibling, round: 'earlier' as const })),
+  ];
+  const collected = await Promise.all(
+    rounds.map(async ({ record: roundRecord, round }) => {
+      const [messages, events] = await Promise.all([
+        store.listCreatorMessages(roundRecord.issueNumber, { limit: MAX_TRANSCRIPT_LIST_ENTRIES }),
+        store.listBuildEvents(roundRecord.issueNumber, { limit: MAX_TRANSCRIPT_LIST_ENTRIES }),
```
[the remark this answers is not in the document]
Comment (gtanczyk) [reply apps/api/src/build-transcript.ts:113]: Confirmed and fixed in 1dc2568. Raised the per-round read ceiling from 50 to 300 (real store pagination against `listCreatorMessages`/`listBuildEvents` would be the fuller fix but felt disproportionate here — neither API supports a `before` cursor today). More importantly, added `truncatedAtSource`, present and `true` only when a round's fetch actually hit the cap, so `hasMore: false` is never presented as "definitely nothing more" when the source read itself was incomplete. Covered in `build-transcript.test.ts` and `agent-channel.test.ts` (the latter against the real store, not a fake).

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (gtanczyk) [reply apps/api/src/build-transcript.ts:113]: Confirmed and fixed in 1dc2568. `collectTranscriptEntries` now synthesizes a `creator_request` entry per round from `roundRecord.spec` (timestamped at the round's own `createdAt`, so it sorts first), skipping it when an improvement round already echoed the identical text as a creator message (`startImprovementRound` does that) so the request doesn't appear to repeat itself. Covered by two new tests in `build-transcript.test.ts` — one for the plain creation-round case, one for the dedup.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by gtanczyk** [comment]
@codex

File: apps/api/src/store.ts [hunk 4]
```diff
@@ -616,6 +616,11 @@ export function isStudioOrigin(origin: CreatorMessageOrigin | undefined): boolea
   return origin === 'studio' || origin === 'studio_ack';
 }
 
+// 1 for the first dispatch; unlike roundGeneration, retries don't bump it.
+export function dispatchAttempt(record: Pick<SubmissionRecord, 'dispatch'>): number {
+  return record.dispatch?.refs?.length ?? 1;
```
[the remark this answers is not in the document]
Comment (gtanczyk) [reply apps/api/src/store.ts]: Confirmed and fixed in f1cb9b2. `dispatchAttempt` is now async and takes `store` + `record`: it sums this job's own `dispatch.refs.length` plus the same-owner sibling jobs for the same slug created earlier (via `listSubmissionsBySlug`), so a fresh improvement-round job's very first dispatch reports the game-wide attempt count instead of resetting to 1. All four call sites (`agent-channel.ts`'s `/brief` route, `mcp-server.ts`'s `start` tool on both the active-round and legacy round-key paths) now `await` it. Covered by a new `describe('dispatchAttempt', ...)` block in `store.test.ts`, including the exact improvement-round regression case (job 2 dispatched once should read 2, not 1).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/api/src/build-transcript.ts [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+): Promise<{ entries: TranscriptEntry[]; truncatedAtSource: boolean }> {
+  const siblings = record.slug
+    ? (await store.listSubmissionsBySlug(record.slug))
+        .filter(
+          (sibling) =>
+            sibling.issueNumber !== record.issueNumber &&
+            sibling.ownerUid === record.ownerUid &&
+            sibling.createdAt < record.createdAt,
+        )
+        .slice(0, MAX_TRANSCRIPT_ROUNDS - 1)
```
[the remark this answers is not in the document]
Comment (gtanczyk) [reply apps/api/src/build-transcript.ts]: Confirmed and fixed in f1cb9b2. `collectTranscriptEntries` now computes the full eligible-sibling list before slicing to `MAX_TRANSCRIPT_ROUNDS - 1`, and sets `truncatedAtSource: true` whenever that slice actually drops any siblings — separate from (and in addition to) the existing per-round entry-cap signal. Covered by two new tests in `build-transcript.test.ts`: one asserting `truncatedAtSource` when a game has one more sibling than the cap keeps, one asserting it's absent when siblings exactly fit the cap.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/api/src/build-transcript.ts [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+      // Skip the synthetic copy when a message already echoed it.
+      const specAlreadyEchoed = trimmedSpec
+        ? messages.some((message) => stripPlaytestContext(message.text).trim() === trimmedSpec)
+        : false;
+      const specEntry: TranscriptEntry[] =
+        trimmedSpec && !specAlreadyEchoed
+          ? [
+              {
+                kind: 'creator_request' as const,
+                text: trimmedSpec.slice(0, MAX_TRANSCRIPT_ENTRY_CHARS),
```
[the remark this answers is not in the document]
Comment (gtanczyk) [reply apps/api/src/build-transcript.ts:147]: Confirmed and fixed in f1cb9b2. Added `SubmissionRecord.specIsSystemGenerated` (and a matching `setSubmissionBrief` param), set by `startImprovementRound` precisely when it's called without `requestedBy` — the case `suggestion-inbox.ts` and `suggestion-sweep.ts` both hit, since their `spec` is a machine-assembled evidence brief, not creator words. `collectTranscriptEntries` now checks this flag when synthesizing a round's spec entry and labels it `agent_note` instead of `creator_request` when set. Covered by a new test in `build-transcript.test.ts` asserting the synthetic entry's kind for a `specIsSystemGenerated` round.

---
_Generated by [Claude Code](https://claude.ai/code)_
